### DISCUSSION TITLE
Delete all relation reference

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [pocketbase/pocketbase#2278] [Q&A] [answered]
Hi,

Quick question:

I have a collection called 'posts' that is related to two other collections namely 'comments' and 'alreadyVoted'. When i now delete the posts i also want to delete the comments and votes associated with it from my database:

`await locals.pb.collection('posts').delete(body.get('postId'));`

How can i do that in a efficient way?

**Comment by ganigeorgiev** [comment, marked as answer]
They will be deleted automatically on `posts` record deletion if:
- your `comments` and `alreadyVoted` collections have `relation` field pointing to the `posts` collection
- the `relation` field has the cascade delete option enabled (aka. the "Delete main record on relation delete" select; it'll be labeled better in the next release).

**Reply by MuslemRahimi** [reply]
thank you
